src/java.base/share/classes/java/io/InputStream.java

```diff
@@ -55,6 +56,32 @@ public abstract class InputStream implements Closeable {
 
     private static final int DEFAULT_BUFFER_SIZE = 8192;
 
+    /**
+     * Only used in {@link #skip(long) skip}, for avoiding multiple times byte array allocation for a same InputStream Object.
+     *
+     * Making this static would increase performance,
+     * but would make it possible to build a malicious subclass of InputStream to steal data from other InputStream Objects.
+     * So DO NOT try to make it static, for the sake of security.
+     */
+    private SoftReference<byte[]> skipBufferReference;
+
+    /**
+     * Get or create a byte array for {@link #skip(long) skip}.
+     *
+     * @param size minimum length that the skip byte array must have.
+     *             notice that this param input MUST be equal or less than {@link #MAX_SKIP_BUFFER_SIZE MAX_SKIP_BUFFER_SIZE}.
+     * @return the byte array.
+     */
+    private byte[] skipBufferReference(int size) {
+        SoftReference<byte[]> ref = this.skipBufferReference;
+        byte[] buffer;
+        if (ref == null || (buffer = ref.get()) == null || buffer.length < size) {
```

Comment:
A comment here or in the method javadoc to the effect that a new buffer is allocated unless the existing buffer is large enough might head off doubt that buffer is always non-null at the return.  As would inverting the if:
```
   if (ref != null && (buffer = ref.get()) != null && buffer.length >= size) {
      return buffer;
   }
   // allocate new or larger buffer
   buffer = new byte[size];
   this.skipBufferReference = new SoftReference<>(buffer);
   return buffer;
```

Reply:
done.